### PR TITLE
Shop page

### PR DESCRIPTION
**Pull request by Vitalia888-170**

### CODE REVIEW

File: spec/requests/shops_spec.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe "Shops", type: :request do
+  describe "GET /shop" do
+    it "returns http success" do
+      get "/shops/shop"
+      expect(response).to have_http_status(:success)
+    end
+  end
+
```
Comment (Shabaldas) [inline spec/requests/shops_spec.rb:10]: ?
empty space.

File: app/helpers/shops_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module ShopsHelper
+end
```
Comment (AndriiBuhalo) [inline app/helpers/shops_helper.rb]: нічого нема всередині видаляй

File: app/controllers/shops_controller.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class ShopsController < ApplicationController
```
Comment (Shabaldas) [inline app/controllers/shops_controller.rb]: Rename Shop => Store

File: app/controllers/shops_controller.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+  def shop
```
Comment (Shabaldas) [inline app/controllers/shops_controller.rb]: shop action => index action

File: app/helpers/shops_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module ShopsHelper
```
Comment (Shabaldas) [inline app/helpers/shops_helper.rb]: remove this file.
And don't create bu scaffold @AndriiBuhalo !!!

File: app/javascript/controllers/priceslider_controller.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    
+    if (toValue <= fromValue) {
+      this.rangeFromTarget.value = toValue
+      this.valueFromTarget.innerHTML = toValue
+    }
+    this.valueToTarget.innerHTML = toValue
+    this.valueToTarget.style.position='relative';
+    this.valueToTarget.style.left=toValue/14 + 'px'
+  }
+
```
Comment (Shabaldas) [inline app/javascript/controllers/priceslider_controller.js:27]: remove empty space

File: app/javascript/controllers/priceslider_controller.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
```
Comment (Shabaldas) [inline app/javascript/controllers/priceslider_controller.js]: remove empty space

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -7,7 +7,7 @@
     <%= csp_meta_tag %>
 
   <link rel="stylesheet" href="https://rsms.me/inter/inter.css">
-    <%= stylesheet_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
+    <%= stylesheet_pack_tag 'shops', 'data-turbolinks-track': 'reload' %>
```
Comment (Shabaldas) [inline app/views/layouts/application.html.erb]: ???
you remove here all our tailwind stylesheets and components

File: app/views/shops/_shop_filter.html.erb [hunk 9]
```diff
@@ -0,0 +1,99 @@
+<div class="pb-4">
+	<h4 class="text-lg text-gray-800">Категорії</h4>
```
Comment (Shabaldas) [inline app/views/shops/_shop_filter.html.erb]: i18n

File: app/views/shops/_shop_filter.html.erb [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+	<div class="mt-3">
+		<div class="flex justify-between">
+			<div class="flex items-center">
+				<input id="medical_models" type="radio" name="category" class="w-2.5 h-2.5 appearance-none bg-white ring-1 ring-offset-2 ring-gray-400 focus:outline-0 checked:bg-yellow-500 bg-none " style="color:#facc15 ; background-image: none; border: none; border-radius: 50%;"/>
+				<label for="medical_models" class="ml-3">Медичні моделі</h5>
```
Comment (Shabaldas) [inline app/views/shops/_shop_filter.html.erb]: move Uranian words to i18n

File: app/views/shops/_shop_filter.html.erb [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+			</div>
+			<p>765</p>
+		</div>
+	</div>
+	<div class="mt-3">
+		<div class="flex justify-between items-center">
+			<div class="flex items-center">
+				<input id="engineer_models" type="radio" name="category" class="w-2.5 h-2.5 appearance-none bg-white ring-1 ring-offset-2 ring-gray-400 focus:outline-0 checked:bg-yellow-500 bg-none " style="color:#facc15 ; background-image: none; border: none; border-radius: 50%;" />
+				<label for="engineer_models" class="ml-3">Інженерні деталі</h5>
```
Comment (Shabaldas) [inline app/views/shops/_shop_filter.html.erb]: here

File: app/views/shops/_shop_filter.html.erb [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+			</div>
+			<p>23</p>
+		</div>
+	</div>
+	<div class="mt-3">
+		<div class="flex justify-between items-center">
+			<div class="flex items-center">
+				<input id="radio_models" type="radio" name="category" class="w-2.5 h-2.5 appearance-none bg-white ring-1 ring-offset-2 ring-gray-400 focus:outline-0 checked:bg-yellow-500 bg-none " style="color:#facc15 ; background-image: none; border: none; border-radius: 50%;" />
+				<label for="radio_models" class="ml-3">Радіо пристрої</h5>
```
Comment (Shabaldas) [inline app/views/shops/_shop_filter.html.erb]: and here

File: app/views/shops/_shop_filter.html.erb [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+			</div>
+			<p>18</p>
+		</div>
+	</div>
+	<div class="mt-3">
+		<div class="flex justify-between items-center">
+			<div class="flex items-center">
+				<input id="new_models" type="radio" name="category" class="w-2.5 h-2.5 appearance-none bg-white ring-1 ring-offset-2 ring-gray-400 focus:outline-0 checked:bg-yellow-500 bg-none " style="color:#facc15 ; background-image: none; border: none; border-radius: 50%;" />
+				<label for="new_models" class="ml-3">Новітні технології</h5>
```
Comment (Shabaldas) [inline app/views/shops/_shop_filter.html.erb]: and here

File: app/views/shops/_shop_filter.html.erb [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+			</div>
+			<p>11</p>
+		</div>
+	</div>
+</div>
+<hr />
+<div class="py-4">
+	<h4 class="text-lg text-gray-800">Ціна</h4>
```
Comment (Shabaldas) [inline app/views/shops/_shop_filter.html.erb]: here

File: app/views/shops/_shop_filter.html.erb [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+				/>
+			<output class="range-slider__value" for="value[from]" data-target="priceslider.valueFrom">0</output>
+			<output class="range-slider__value" for="value[to]" data-target="priceslider.valueTo">0</output>
+		</fieldset>
+	</div>
+	
+</div>
+<hr />
+<div class="py-4">
+	<h4 class="text-lg text-gray-800">Матеріали</h4>
```
Comment (Shabaldas) [inline app/views/shops/_shop_filter.html.erb]: here

File: app/views/shops/_shop_filter.html.erb [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+	<div class="mt-3">
+		<div class="flex items-center">
+			<input id="material1" type="radio" name="material" class="w-2.5 h-2.5 appearance-none bg-white ring-1 ring-offset-2 ring-gray-400 focus:outline-0 checked:bg-yellow-500 bg-none " style="color:#facc15 ; background-image: none; border: none; border-radius: 50%;" />
+			<label for="material1" class="ml-3">Пластмаса 32</h5>
+		</div>
+	</div>
+	<div class="mt-3">
+		<div class="flex items-center">
+			<input id="material2" type="radio" name="material" class="w-2.5 h-2.5 appearance-none bg-white ring-1 ring-offset-2 ring-gray-400 focus:outline-0 checked:bg-yellow-500 bg-none " style="color:#facc15 ; background-image: none; border: none; border-radius: 50%;" />
+			<label for="material2" class="ml-3">Пластмаса 18</h5>
```
Comment (Shabaldas) [inline app/views/shops/_shop_filter.html.erb]: here

File: app/views/shops/_shop_filter.html.erb [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+		</div>
+	</div>
+	<div class="mt-3">
+		<div class="flex items-center">
+			<input id="material3" type="radio" name="material" class="w-2.5 h-2.5 appearance-none bg-white ring-1 ring-offset-2 ring-gray-400 focus:outline-0 checked:bg-yellow-500 bg-none " style="color:#facc15 ; background-image: none; border: none; border-radius: 50%;" />
+			<label for="material3" class="ml-3">Нова пластмаса</h5>
```
Comment (Shabaldas) [inline app/views/shops/_shop_filter.html.erb]: here

File: app/views/shops/shop.html.erb [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+					<h4 class="text-white text-2xl font-medium">Organic Cubes</h4>
+					<p class="text-3xl text-white pt-5">₴ 329</p>
+				</div>
+				<div class="w-7/12 lg:w-10/12 absolute bottom-0 right-0">
+					<%= image_tag("shop_img/model5.png") %>
+				</div>
+			</div>
+		</div>
+	</div>
+</div>
```
Comment (Shabaldas) [inline app/views/shops/shop.html.erb]: i18n in all this file

File: spec/helpers/shops_helper_spec.rb [hunk 19]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
```
Comment (Shabaldas) [inline spec/helpers/shops_helper_spec.rb]: remove all this file

File: spec/requests/shops_spec.rb [hunk 20]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
```
Comment (Shabaldas) [inline spec/requests/shops_spec.rb]: remove all this file.
We don’t have specs file yet

File: spec/views/shops/shop.html.slim_spec.rb [hunk 21]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (Shabaldas) [inline spec/views/shops/shop.html.slim_spec.rb]: remove this file

**Review by Shabaldas — CHANGES_REQUESTED** [review]

**Comment by Shabaldas** [comment]
1. Rename Shop to Store.
2. In Store we use default index action not ''shop'' or ''store''
3. In all .rb files change " => '
4. Connect custom css file to javascript => stylesheets => application.scss
5. Remove unnecessary files

File: spec/requests/shops_spec.rb (hunk 1 above)
Comment (AndriiBuhalo) [inline spec/requests/shops_spec.rb]: all is correct empty line in the end is mast be

Comment (AndriiBuhalo) [reply spec/requests/shops_spec.rb]: https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

File: app/javascript/controllers/priceslider_controller.js [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
```
Comment (Shabaldas) [inline app/javascript/controllers/priceslider_controller.js]: why you don’t use 1 line?

File: app/javascript/controllers/priceslider_controller.js [hunk 23]
```diff
@@ -0,0 +1,44 @@
...
+      this.rangeFromTarget.value = toValue
+      this.valueFromTarget.innerHTML = toValue
+    }
+    this.valueToTarget.innerHTML = toValue
+    this.valueToTarget.style.position='relative';
+    this.valueToTarget.style.left=toValue/14 + 'px'
+  }
+
+  
+}
```
Comment (Shabaldas) [inline app/javascript/controllers/priceslider_controller.js]: add empty space

File: config/routes.rb [hunk 24]
```diff
@@ -6,6 +6,7 @@
   devise_for :users, controllers: { omniauth_callbacks: 'users/omniauth_callbacks' }
   root 'static_pages#home', as: :home
 
+  get 'shop', to: 'shops#shop', as: :shop
```
Comment (Shabaldas) [inline config/routes.rb]: change action routes to
```resources :store, only: :index```

**Review by Shabaldas — CHANGES_REQUESTED** [review]

File: app/helpers/shops_helper.rb (hunk 2 above)
Comment (Vitalia888-170) [reply app/helpers/shops_helper.rb]: fixed

File: app/controllers/shops_controller.rb (hunk 3 above)
Comment (Vitalia888-170) [reply app/controllers/shops_controller.rb]: done

File: app/controllers/shops_controller.rb (hunk 4 above)
Comment (Vitalia888-170) [reply app/controllers/shops_controller.rb]: done

File: app/helpers/shops_helper.rb (hunk 5 above)
Comment (Vitalia888-170) [reply app/helpers/shops_helper.rb]: okay, done

File: app/javascript/controllers/priceslider_controller.js (hunk 6 above)
Comment (Vitalia888-170) [reply app/javascript/controllers/priceslider_controller.js:27]: fixed

File: app/javascript/controllers/priceslider_controller.js (hunk 7 above)
Comment (Vitalia888-170) [reply app/javascript/controllers/priceslider_controller.js]: fixed

File: app/views/layouts/application.html.erb (hunk 8 above)
Comment (Vitalia888-170) [reply app/views/layouts/application.html.erb]: Sorry, some accident. Done

File: app/views/shops/_shop_filter.html.erb (hunk 9 above)
Comment (Vitalia888-170) [reply app/views/shops/_shop_filter.html.erb]: done

File: app/views/shops/_shop_filter.html.erb (hunk 10 above)
Comment (Vitalia888-170) [reply app/views/shops/_shop_filter.html.erb]: done

File: app/views/shops/_shop_filter.html.erb (hunk 11 above)
Comment (Vitalia888-170) [reply app/views/shops/_shop_filter.html.erb]: done

File: app/views/shops/_shop_filter.html.erb (hunk 12 above)
Comment (Vitalia888-170) [reply app/views/shops/_shop_filter.html.erb]: done

File: app/views/shops/_shop_filter.html.erb (hunk 13 above)
Comment (Vitalia888-170) [reply app/views/shops/_shop_filter.html.erb]: done

File: app/views/shops/_shop_filter.html.erb (hunk 14 above)
Comment (Vitalia888-170) [reply app/views/shops/_shop_filter.html.erb]: done

File: app/views/shops/_shop_filter.html.erb (hunk 15 above)
Comment (Vitalia888-170) [reply app/views/shops/_shop_filter.html.erb]: done

File: app/views/shops/_shop_filter.html.erb (hunk 16 above)
Comment (Vitalia888-170) [reply app/views/shops/_shop_filter.html.erb]: done

File: app/views/shops/_shop_filter.html.erb (hunk 17 above)
Comment (Vitalia888-170) [reply app/views/shops/_shop_filter.html.erb]: done

File: app/views/shops/shop.html.erb (hunk 18 above)
Comment (Vitalia888-170) [reply app/views/shops/shop.html.erb]: done

File: spec/helpers/shops_helper_spec.rb (hunk 19 above)
Comment (Vitalia888-170) [reply spec/helpers/shops_helper_spec.rb]: done

File: spec/requests/shops_spec.rb (hunk 20 above)
Comment (Vitalia888-170) [reply spec/requests/shops_spec.rb]: done

File: spec/views/shops/shop.html.slim_spec.rb (hunk 21 above)
Comment (Vitalia888-170) [reply spec/views/shops/shop.html.slim_spec.rb]: done

File: app/javascript/controllers/priceslider_controller.js (hunk 22 above)
Comment (Vitalia888-170) [reply app/javascript/controllers/priceslider_controller.js]: fixed

File: app/javascript/controllers/priceslider_controller.js (hunk 23 above)
Comment (Vitalia888-170) [reply app/javascript/controllers/priceslider_controller.js]: fixed

File: config/routes.rb (hunk 24 above)
Comment (Vitalia888-170) [reply config/routes.rb]: thanks for piece of advice

File: app/assets/config/manifest.js [hunk 25]
```diff
@@ -1,2 +1,4 @@
 //= link_tree ../images
+//= link_directory ../stylesheets .css
```
Comment (Shabaldas) [inline app/assets/config/manifest.js]: remove this link

**Review by Shabaldas — APPROVED** [review]

**Review by Shabaldas — APPROVED** [review]

**Review by AndriiBuhalo — APPROVED** [review]

**Pull request merged**